### DISCUSSION TITLE
Stamp annotation not being displayed in Acrobat

### DISCUSSION THREAD

**Discussion by tiago719** [py-pdf/pypdf#1670] [Show and tell]
I am currently developing a script that iterates over the pages of a PDF file, removes specified annotations, and generates a new PDF file as output.
To simplify the example presented below, I have only included code for iterating over the pages and writing them to the new PDF
 Upon opening the resulting "new_sample.pdf" file with Acrobat, it can be observed that the stamp annotation that existed in the original file has been unexpectedly removed.

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from pypdf import PdfReader, PdfWriter

reader = PdfReader('sample.pdf')
writer = PdfWriter()
for page in reader.pages:
    # Some operations that will change the page object
    writer.add_page(page.get_object())

with open('new_sample.pdf', 'w+b') as f:
    writer.write(f.name)

```

[sample.pdf](https://github.com/py-pdf/pypdf/files/10842846/sample.pdf)

**Comment by pubpub-zz** [comment]
I recommend to use append() instead that add_page() that is pulling more data
as you are copying the whole document this will be more simple:

```
from pypdf import PdfWriter

writer = PdfWriter()
writer.append('sample.pdf')
writer.write('new_sample.pdf')
````

Can you check you are getting what you are expecting ?

**Comment by tiago719** [comment]
In order to modify the contents of the pages, a new PDF document cannot be created based solely on the original. As a result, I have to iterate through the pages, conduct relevant operations on the page object, and subsequently save the adjusted pages to a new PDF file.

**Comment by pubpub-zz** [comment]
oups to quick/tired 🥴. I've updated the code propolsa

**Comment by pubpub-zz** [comment]
> In order to modify the contents of the pages, a new PDF document cannot be created based solely on the original. As a result, I have to iterate through the pages, conduct relevant operations on the page object, and subsequently save the adjusted pages to a new PDF file.

Once the page have been copied in, can your code modify them ?
you can use 
`for page in writer.pages`

**Comment by pubpub-zz** [comment]
I close this issue and convert it as a discussion.

**Comment by tiago719** [comment]
> I recommend to use append() instead that add_page() that is pulling more data as you are copying the whole document this will be more simple:
> 
> ```
> from pypdf import PdfWriter
> 
> writer = PdfWriter()
> writer.append('sample.pdf')
> writer.write('new_sample.pdf')
> ```
> 
> Can you check you are getting what you are expecting ?

This approach works, however I still have the same result in the new file - the stamp annotation is not there.

**Comment by pubpub-zz** [comment]
I've been a little to quick to convert it to discussion. I create an issue on last post
